region/src/main/java/org/motechproject/nms/region/service/impl/VillageServiceImpl.java

```diff
@@ -47,12 +47,14 @@ public Village execute(Query query) {
     }
 
     @Override
-    public void create(Village village) {
-        dataService.create(village);
+    public Village create(Village village) {
+        Village v = dataService.create(village);
+        dataService.evictEntityCache(true);
```

Comment:
Why does village evict the cache but none of the other location entities evict the cache?